src/types.rs

```diff
@@ -1,9 +1,80 @@
 use crate::errors;
-use std::{iter::FromIterator, result};
+use std::{
+    iter::FromIterator,
+    result,
+    time::{Duration, SystemTime, UNIX_EPOCH},
+};
 
 /// A time tag in OSC message consists of two 32-bit integers where the first one denotes the number of seconds since 1900-01-01 and the second the fractions of a second.
 /// For details on its semantics see http://opensoundcontrol.org/node/3/#timetags
-pub type OscTime = (u32, u32);
+///
+/// # Examples
+///
+/// ```
+/// use rosc::OscTime;
+/// use std::time::SystemTime;
+///
+/// assert_eq!(OscTime::from(SystemTime::UNIX_EPOCH), OscTime::from((2_208_988_800, 0)));
+/// ```
+///
+/// # Conversions between [`std::time::SystemTime`]
+///
+/// The [`From`]/[`Into`] traits are implemented to allow conversions between
+/// [`SystemTime`](std::time::SystemTime) and `OscTime` in both directions. **These conversions are
+/// lossy**, but are tested to have a deviation within 5 nanoseconds when converted back and forth
+/// in either direction.
+#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub struct OscTime {
+    pub seconds: u32,
+    pub fractional: u32,
+}
+
+impl OscTime {
+    const UNIX_OFFSET: u64 = 2_208_988_800; // From RFC 5905
+    const TWO_POW_32: f64 = 4294967296.0; // Number of bits in a `u32`
+    const NANOS_PER_SECOND: f64 = 1.0e9;
+
+    fn epoch() -> SystemTime {
+        UNIX_EPOCH - Duration::from_secs(OscTime::UNIX_OFFSET)
+    }
+}
+
+impl From<SystemTime> for OscTime {
+    fn from(time: SystemTime) -> OscTime {
+        let duration_since_epoch = time.duration_since(OscTime::epoch()).unwrap();
```

Comment:
Instead of returning a default value here, I changed the implementation to use the `TryFrom` trait and pass the error up. This seems more in line with how `SystemTime` itself handles errors. I also noticed the conversion here from seconds here from a `u64` to a `u32` could silently overflow overflow so I made an error type and returned it there.